snare/cloner.py

```diff
@@ -21,7 +22,18 @@
 
 
 class BaseCloner:
-    def __init__(self, root, max_depth, css_validate, default_path="/opt/snare"):
+    def __init__(self, root: str, max_depth: int, css_validate: bool, default_path: str = "/opt/snare") -> None:
+        """Base class for all core functions of the cloner
+
+        :param root: Website root URL
+        :type root: str
```

Comment:
do we need type if the type annotated in function definition?

Reply:
Yes, Sphinx [mandates it](https://sphinx-rtd-tutorial.readthedocs.io/en/latest/docstrings.html#the-sphinx-docstring-format) and the type disappears from the parameter description in docs when we remove its `:type` from the docstring.